third_party/pyth/price-service/src/rest.ts

```diff
@@ -113,6 +120,55 @@ export class RestAPI {
       "api/latest_vaas?ids[]=<price_feed_id>&ids[]=<price_feed_id_2>&.."
     );
 
+    const getVaaInputSchema: schema = {
+      query: Joi.object({
+        id: Joi.string()
+          .regex(/^(0x)?[a-f0-9]{64}$/)
+          .required(),
+        // publishTime is of UNIX timestamp type
+        publish_time: Joi.number().required(),
+      }).required(),
+    };
+    app.get(
+      "/api/get_vaa",
+      validate(getVaaInputSchema),
+      (req: Request, res: Response) => {
+        const priceFeedId = req.query.id as string;
+        const publishTime = Number(req.query.publish_time as string);
+        const vaa = this.priceFeedVaaInfo.getVaa(priceFeedId, publishTime);
+        // if publishTime is older than cache ttl or vaa is not found, fetch from db
+        if (
+          publishTime <
+            Math.floor(Date.now() / 1000) -
+              PYTH_WRITER_INSERT_LATENCY_SECONDS ||
+          !vaa
+        ) {
+          // cache miss
+          fetch(
+            `https://web-api.pyth.network/vaa?id=${priceFeedId}&publishTime=${publishTime}&cluster=pythnet`
```

Comment:
I would make the endpoint an Env var and if it's not given I won't query it. Because I'm thinking of not exposing it publicly.

Reply:
Also, the cluster is ambiguous here. We should support devnet too. I suggest `testnet` and `mainnet` as that is how we call our price service clusters (and price service works regardless of pythnet/solana)

Reply:
`mainnet` actually refers to `pythnet` for web-api endpoint, are you suggesting we should have a mapping instead that maps mainnet -> pythnet, and testnet -> pythtest?